src/inject.js

```diff
@@ -1,41 +1,33 @@
 import React, { Component, createElement } from "react"
 import { observer } from "./observer"
-import { isStateless, copyStaticProperties } from "./utils/utils"
+import { copyStaticProperties } from "./utils/utils"
 import { MobXProviderContext } from "./Provider"
 
 /**
  * Store Injection
  */
 function createStoreInjector(grabStoresFn, component, injectNames, makeReactive) {
-    let displayName = getInjectName(component, injectNames)
-
-    class Injector extends Component {
-        static contextType = MobXProviderContext
-
-        render() {
-            const { forwardRef, ...props } = this.props
+    // Support forward refs
+    let Injector = React.forwardRef((props, ref) => {
+        const newProps = { ...props }
+        const context = React.useContext(MobXProviderContext)
+        Object.assign(newProps, grabStoresFn(context || {}, newProps) || {})
 
-            Object.assign(props, grabStoresFn(this.context || {}, props) || {})
+        if (ref) {
```

Comment:
I wonder if that condition is actually needed. Worst case scenario the `undefined` gets assigned here and that's like nothing happened. That means you can just stick that ref to line 12, right?

Reply:
We cannot do it because it will be `null`, not `undefined` and it will trigger the following warning for function components:
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```